reconcile/ocm_clusters.py

```diff
@@ -24,6 +24,42 @@ def fetch_desired_state(clusters):
     return desired_state
 
 
+def get_cluster_update_spec(cluster_name, current_spec, desired_spec):
+    """ Get a cluster spec to update. Returns an error if diff is invalid """
+    allowed_spec_update_fields = [
+        'instance_type',
+        'storage',
+        'load_balancers',
+        'private',
+        'channel',
+        'autoscale',
+        'nodes'
+    ]
+    error = False
+    if current_spec['network'] != desired_spec['network']:
+        error = True
+        logging.error(f'[{cluster_name}] invalid update: network')
+    current_spec_spec = current_spec['spec']
+    desired_spec_spec = desired_spec['spec']
+    updated = {k: desired_spec_spec[k] for k in desired_spec_spec
+               if current_spec_spec[k] != desired_spec_spec[k]}
+
+    # we only need deleted to check if a field removal is valid
+    # we really want to check updated + deleted, and since
+    # we have no further use for deleted -
+    deleted = {k: current_spec_spec[k] for k in current_spec_spec
+               if k not in desired_spec_spec}
+    diffs = deleted
+    diffs.update(updated)
+
+    for k in diffs:
+        if k not in allowed_spec_update_fields:
+            error = True
```

Comment:
Why not raise `ValueError` instead? This boolean flag feels clunky

Reply:
it's some golang-like error handling which we do from time to time instead of raising, catching, logging error, continuing.

Reply:
That's my point, golang style is best left for golang.